README.md

```diff
@@ -12,30 +15,107 @@ is simpler and easier to work with. Additionally, for developers familar with
 the Go programming language will be instantly comfortable with Gunk files and
 syntax.
 
+### Contents
+
+* [How it works](#how-it-works)
+* [Features](#features)
+* [Installing](#installing)
+* [Usage](#usage)
+* [Contributing](#contributing)
+
+## How it works
+
+Gunk works by using Go's standard library `go/ast` package (and related
+packages) to parse all `*.gunk` files in a "Gunk Package" (and applying the
+same kinds of rules as Go package), building the appropos protobuf messages. In
+turn, those are passed to `protoc-gen-*` tools, along with any passed options.
+
+## Features
+
+- Written in simple and idiomatic [Go][go-project]
+- Intuitive and [easy to use](#usage)
+- Works on Linux, Mac and Windows
+
 ## Installing
 
-Gunk can be installed in the usual Go fashion:
+Gunk requires [Go][go-project] to be installed on your machine. To install Go,
+follow the [installation instructions][go-install] for your specific operating
+systems.
+
+Gunk can then be installed in the usual Go fashion:
+
+	$ go get -u github.com/gunk/gunk
```

Comment:
I personally slightly prefer leaving out the `$` when we're just giving commands for people to run. Otherwise copy-pasting is a bit more difficult, as one has to be careful not to copy the dollar.

I do use dollars when showcasing how something works though, to differentiate the prompt from the output:

```
$ gunk awesome-stuff
Wow such awesome stdout!
```